code/game/objects/structures/inflatable.dm

```diff
@@ -475,3 +475,9 @@
 		user.visible_message("<span class='danger'>[user] begins to drag [target] into the shelter!</span>")
 		if(do_after_many(user,list(target,src),20)) //Twice the normal time
 			enter_shelter(target)
+
+/obj/structure/inflatable/shelter/Exited(var/atom/movable/mover)
+	if (mover in contents)
```

Comment:
Is this tested? Because this should always be false.

Reply:
https://webmshare.com/play/n5zVL

Author's note : `contents -= mover` is needed, because else, the phantom sprite is still in the shelter.

Reply:
And `forceMove()` sets `loc` *before* it calls `Exited()`

Reply:
`forceMove` removing the thing from the contents _before_ calling `Exited()` is the cleanest way I can think of doing it.